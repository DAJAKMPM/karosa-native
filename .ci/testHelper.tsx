import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
