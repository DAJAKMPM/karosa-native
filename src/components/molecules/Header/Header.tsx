/**
 *
 * Header
 * @format
 *
 */

import React, { FC } from "react";
import { Header as RnHeader } from "react-native-elements";

import type { PropsType } from "./types";
import InputStyles from "./styles";

const Header: FC<PropsType> = (props) => {
  const {
    children,
    centerComponent,
    customStyle,
    isCustom,
    leftComponent,
    placement,
    rightComponent,
    barStyle,
  } = props;

  if (isCustom) {
    return <RnHeader>{children}</RnHeader>;
  }

  return (
    <RnHeader
      placement={placement}
      barStyle={barStyle}
      containerStyle={[InputStyles.container, customStyle]}
      leftComponent={leftComponent}
      centerComponent={centerComponent}
      rightComponent={rightComponent}
    />
  );
};

export default Header;
