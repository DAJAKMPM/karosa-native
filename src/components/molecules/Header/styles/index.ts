/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

const HeaderStyles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.transparent,
  },
});

export default HeaderStyles;
