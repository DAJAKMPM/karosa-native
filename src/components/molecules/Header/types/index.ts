/**
 *
 * @format
 *
 */

import { StyleProp, ViewStyle } from "react-native";
import { HeaderSubComponent } from "react-native-elements";

type Placement = "left" | "center" | "right";

type BarStyle = "default" | "light-content" | "dark-content";

export type PropsType = {
  isCustom?: boolean;
  barStyle?: BarStyle;
  placement?: Placement;
  containerStyle?: StyleProp<ViewStyle>;
  leftComponent?: HeaderSubComponent;
  centerComponent?: HeaderSubComponent;
  rightComponent?: HeaderSubComponent;
};
