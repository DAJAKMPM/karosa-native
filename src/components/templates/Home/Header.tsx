/**
 *
 * Header
 * @format
 *
 */

import React, { FC, memo, useMemo } from "react";
import { ImageBackground, View } from "react-native";
import BaseHeader from "@app/molecules/Header";
import Searchbar from "@app/molecules/SearchBar";
import Icon from "@app/atoms/Icon";

import { HeaderStyles, HomeStyles } from "./styles";
import { ICON_SIZE, BLUR_RADIUS } from "./config";

const Header: FC = () => {
  const containerStyle = useMemo(() => [HeaderStyles.container], []);

  return (
    <View style={containerStyle}>
      <ImageBackground
        style={HeaderStyles.coverPhoto}
        blurRadius={BLUR_RADIUS}
        source={{
          uri: "https://res.cloudinary.com/dyfla7mxr/image/upload/v1614606613/karosa/shop_ynswwn.jpg",
        }}
      >
        <BaseHeader
          containerStyle={HeaderStyles.headerContainer}
          barStyle="light-content"
          leftComponent={
            <Searchbar
              backgroundColor="primary"
              size="sm"
              placeholder="What are you looking for?"
            />
          }
          rightComponent={
            <View style={HomeStyles.horizontalContainer}>
              <Icon
                group="home"
                name="cart"
                height={ICON_SIZE.header}
                width={ICON_SIZE.header}
                extraStyle={HeaderStyles.cartIcon}
              />
              <Icon
                group="home"
                name="chat"
                height={ICON_SIZE.header}
                width={ICON_SIZE.header}
              />
            </View>
          }
        />
      </ImageBackground>
    </View>
  );
};

export default memo(Header);
