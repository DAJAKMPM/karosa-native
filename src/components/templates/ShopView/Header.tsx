/**
 *
 * Header
 * @format
 *
 */

import React, { FC, Fragment, memo, useMemo } from "react";
import { ImageBackground, StatusBar, View } from "react-native";
import { ICON } from "@app/organisms/Profile/config";
import BaseHeader from "@app/molecules/Header";
import Searchbar from "@app/molecules/SearchBar";
import Image from "@app/atoms/Image";
import Text from "@app/atoms/Text";
import Button from "@app/atoms/Button";
import Icon from "@app/atoms/Icon";

import type { HeaderProps } from "./types";
import { HeaderStyles } from "./styles";

const Header: FC<HeaderProps> = (props) => {
  const { onBack, style } = props;

  const containerStyle = useMemo(() => [HeaderStyles.container, style], []);

  return (
    <View style={containerStyle}>
      <ImageBackground
        style={HeaderStyles.coverPhoto}
        blurRadius={1}
        source={{
          uri: "https://res.cloudinary.com/dyfla7mxr/image/upload/v1614606613/karosa/shop_ynswwn.jpg",
        }}
      >
        <StatusBar translucent backgroundColor="transparent" />
        <BaseHeader
          containerStyle={HeaderStyles.headerContainer}
          barStyle="light-content"
          leftComponent={{
            icon: "arrow-back",
            color: "white",
            onPress: onBack,
          }}
          centerComponent={
            <Searchbar
              backgroundColor="primary"
              size="sm"
              placeholder="Search in Shop"
            />
          }
          rightComponent={{
            icon: "more-horiz",
            color: "white",
          }}
        />
      </ImageBackground>
      <View style={HeaderStyles.avatarContainer}>
        <Image
          imageStyle={HeaderStyles.avatarPhoto}
          source={{
            uri: "https://res.cloudinary.com/dyfla7mxr/image/upload/v1614606614/karosa/hinata_dm5sdk.png",
          }}
          resizeMode="contain"
        />
      </View>
      <View style={HeaderStyles.buttonMainContainer}>
        <View style={HeaderStyles.buttonSubContainer}>
          <View style={HeaderStyles.buttonContainer}>
            <Button
              icon={
                <Icon
                  group="common"
                  name="chat"
                  width={ICON.BUTTON}
                  height={ICON.BUTTON}
                  extraStyle={HeaderStyles.chatIcon}
                />
              }
              titleStyle={HeaderStyles.txtChat}
              buttonStyle={HeaderStyles.btnChat}
              title={"Chat"}
            />
          </View>
          <View style={HeaderStyles.buttonContainer}>
            <Button
              icon={
                <Icon
                  group="products"
                  name="add"
                  width={ICON.BUTTON}
                  height={ICON.BUTTON}
                  extraStyle={HeaderStyles.personIcon}
                />
              }
              titleStyle={HeaderStyles.txtPerson}
              buttonStyle={HeaderStyles.btnPerson}
              title={"Follow"}
            />
          </View>
        </View>
        <View style={HeaderStyles.shopNameContainer}>
          <Text
            text="Mercado de Karosa Shop"
            textStyle={HeaderStyles.txtShopName}
          />
        </View>
        <View style={HeaderStyles.shopAddrContainer}>
          <Text
            text="166, Apas Lahug Cebu"
            textStyle={HeaderStyles.txtShopAddr}
          />
          <View style={HeaderStyles.activeContainer}>
            <Fragment>
              <Icon
                group="common"
                name={"greenCircle"}
                height={8}
                width={8}
                extraStyle={HeaderStyles.activeIcon}
              />
              <Text text={"Active"} textStyle={HeaderStyles.txtIcon} />
            </Fragment>
          </View>
        </View>
        <View style={HeaderStyles.bottomInfoContainer}>
          <View style={HeaderStyles.ratingContainer}>
            <Text text={`2.0 / 5.0`} textStyle={HeaderStyles.txtBottomValue} />
            <Text text="Shop Rating" textStyle={HeaderStyles.txtBottomLabel} />
          </View>
          <View style={HeaderStyles.followersContainer}>
            <Text text="4.3K" textStyle={HeaderStyles.txtBottomValue} />
            <Text text="Followers" textStyle={HeaderStyles.txtBottomLabel} />
          </View>
          <View style={HeaderStyles.chatPerformanceContainer}>
            <Text
              text="89%"
              textStyle={[
                HeaderStyles.txtBottomValue,
                HeaderStyles.txtChatPerf,
              ]}
            />
            <Text
              text=" Chat Performance"
              textStyle={HeaderStyles.txtBottomLabel}
            />
          </View>
        </View>
      </View>
    </View>
  );
};

export default memo(Header);
