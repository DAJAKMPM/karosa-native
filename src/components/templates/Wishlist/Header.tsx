import Header from "@app/components/molecules/Header";
import { theme } from "@app/styles";
import React, { FC } from "react";
import type { HeaderProps } from "./types";

const WishlistHeader: FC<HeaderProps> = (props) => {
  const { products } = props;

  return (
    <Header
      hasBottomDivider
      leftComponent={{
        icon: "arrow-back",
        color: theme.colors.primary,
        // onPress: navigation.onBack,
      }}
      centerComponent={{
        text: "My Wishlist" + "(" + products.length + ")",
      }}
      rightComponent={{
        icon: "chat",
        color: theme.colors.primary,
      }}
    />
  );
};

export default WishlistHeader;
