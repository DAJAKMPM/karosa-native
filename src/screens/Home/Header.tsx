import { DIMENS, SPACING, theme } from "@app/styles";
import React, { FC, Fragment, memo, useMemo } from "react";
import {
  ImageBackground,
  ImageProps,
  StatusBar,
  StyleSheet,
  View,
  ViewProps,
} from "react-native";
import BaseHeader from "@app/molecules/Header";
import Searchbar from "@app/molecules/SearchBar";
import Image from "@app/atoms/Image";
import { ICON } from "@app/components/organisms/Profile/config";
import Text from "@app/atoms/Text";
import Button from "@app/atoms/Button";

import Icon from "@app/atoms/Icon";

export const PHOTO_SIZE = 120;

type Props = Pick<ViewProps, "style"> & {
  photo: string;
  name: string;
  bio: string;
};

const Header: FC<Props> = ({ style, name, photo, bio }) => {
  const containerStyle = useMemo(() => [styles.container, style], []);

  const photoSource = useMemo<ImageProps["source"]>(() => ({ uri: photo }), []);

  return (
    <View style={containerStyle}>
      <ImageBackground
        style={styles.coverPhoto}
        blurRadius={1}
        source={{
          uri:
            "https://res.cloudinary.com/dyfla7mxr/image/upload/v1614606613/karosa/shop_ynswwn.jpg",
        }}>
        <StatusBar translucent backgroundColor="transparent" />
        <BaseHeader
          containerStyle={styles.headerContainer}
          barStyle="light-content"
          leftComponent={{
            icon: "arrow-back",
            color: "white",
          }}
          centerComponent={
            <Searchbar
              backgroundColor="primary"
              size="sm"
              placeholder="Search in Shop"
            />
          }
          rightComponent={{
            icon: "more-horiz",
            color: "white",
          }}
        />
      </ImageBackground>
      <View style={styles.avatarContainer}>
        <Image
          imageStyle={styles.avatarPhoto}
          source={{
            uri:
              "https://res.cloudinary.com/dyfla7mxr/image/upload/v1614606614/karosa/hinata_dm5sdk.png",
          }}
          resizeMode="contain"
        />
      </View>
      <View
        style={{
          flex: 1,
          width: DIMENS.screenWidth,
          flexDirection: "column",
          paddingLeft: 16,
          paddingRight: 16,
        }}>
        <View
          style={{
            flexDirection: "row",
            justifyContent: "flex-end",
          }}>
          <View style={{ width: 100, marginRight: 10, marginTop: 15 }}>
            <Button
              icon={
                <Icon
                  group="common"
                  name="chat"
                  width={ICON.BUTTON}
                  height={ICON.BUTTON}
                  extraStyle={styles.chatIcon}
                />
              }
              titleStyle={styles.txtChat}
              buttonStyle={styles.btnChat}
              title={"Chat"}
            />
          </View>
          <View style={{ width: 100, marginRight: 10, marginTop: 15 }}>
            <Button
              icon={
                <Icon
                  group="products"
                  name="add"
                  width={ICON.BUTTON}
                  height={ICON.BUTTON}
                  extraStyle={styles.personIcon}
                />
              }
              titleStyle={styles.txtPerson}
              buttonStyle={styles.btnPerson}
              title={"Follow"}
            />
          </View>
        </View>
        <View style={styles.shopNameContainer}>
          <Text text="Mercado de Karosa Shop" textStyle={styles.txtShopName} />
        </View>
        <View style={styles.shopAddrContainer}>
          <Text text="166, Apas Lahug Cebu" textStyle={styles.txtShopAddr} />
          <View style={styles.activeContainer}>
            <Fragment>
              <Icon
                group="common"
                name={"greenCircle"}
                height={8}
                width={8}
                extraStyle={styles.activeIcon}
              />
              <Text text={"Active"} textStyle={styles.txtIcon} />
            </Fragment>
          </View>
        </View>
        <View style={styles.bottomInfoContainer}>
          <View style={styles.ratingContainer}>
            <Text text={`2.0 / 5.0`} textStyle={styles.txtBottomValue} />
            <Text text="Shop Rating" textStyle={styles.txtBottomLabel} />
          </View>
          <View style={styles.followersContainer}>
            <Text text="4.3K" textStyle={styles.txtBottomValue} />
            <Text text="Followers" textStyle={styles.txtBottomLabel} />
          </View>
          <View style={styles.chatPerformanceContainer}>
            <Text
              text="89%"
              textStyle={[styles.txtBottomValue, { marginLeft: 4 }]}
            />
            <Text text=" Chat Performance" textStyle={styles.txtBottomLabel} />
          </View>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    backgroundColor: theme.colors.white,
    height: 300,
  },
  avatarPhoto: {
    width: DIMENS.screenHeight * 0.15,
    height: DIMENS.screenHeight * 0.15,
    borderRadius: (DIMENS.screenHeight * 0.15) / 2,
    borderColor: theme.colors.white,
    borderWidth: 5,
  },
  avatarContainer: {
    position: "absolute",
    top: DIMENS.screenHeight * 0.14,
    left: DIMENS.screenWidth * 0.038,
    zIndex: 1,
  },
  coverPhoto: {
    height: DIMENS.screenHeight * 0.22,
  },
  headerContainer: { backgroundColor: theme.colors.transparent },
  chatIcon: {
    position: "absolute",
    left: 25,
  },
  txtChat: {
    ...theme.textLight,
    alignSelf: "center",
    position: "relative",
    left: 10,
    color: theme.colors.dark20,
    fontWeight: "700",
  },
  btnChat: {
    height: 32,
    borderRadius: 3,
    backgroundColor: theme.colors.light10,
  },
  personIcon: {
    position: "absolute",
    left: DIMENS.screenWidth * 0.05,
  },
  txtPerson: {
    ...theme.textLight,
    alignSelf: "center",
    position: "relative",
    left: DIMENS.screenWidth * 0.02,
    fontWeight: "700",
  },
  btnPerson: {
    height: 32,
    borderRadius: 3,
  },
  shopNameContainer: {
    marginTop: 10,
  },
  txtShopName: {
    ...theme.textBold,
    fontWeight: "500",
  },
  shopAddrContainer: {
    marginTop: 10,
    flexDirection: "row",
  },
  txtShopAddr: {
    ...theme.textLight,
    fontWeight: "300",
  },
  activeContainer: {
    flexDirection: "row",
    marginLeft: SPACING.md,
  },
  activeIcon: {
    marginTop: SPACING.xs - 1,
  },
  txtIcon: {
    ...theme.textLightBold,
    marginLeft: SPACING.xs,
    fontWeight: "700",
  },
  bottomInfoContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    marginTop: 10,
  },
  ratingContainer: {
    flexDirection: "column",
  },
  txtBottomValue: {
    ...theme.textLightBold,
    fontWeight: "700",
  },
  txtBottomLabel: {
    ...theme.textLight,
    fontWeight: "300",
  },
  followersContainer: {
    flexDirection: "column",
  },
  chatPerformanceContainer: {
    flexDirection: "column",
  },
});

export default memo(Header);
