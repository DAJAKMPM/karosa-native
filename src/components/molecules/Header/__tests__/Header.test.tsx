import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Header from "../Header";

describe("Header Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<Header />);
    expect(toJSON()).toMatchSnapshot();
  });
});
