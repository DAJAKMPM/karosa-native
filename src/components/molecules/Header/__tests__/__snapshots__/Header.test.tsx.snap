// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
</View>
`;
